Building Mono w/ LLVM runtime on ARM64 host fails

Built with `./build.sh -c Release /p:MonoEnableLlvm=true`  (ie. implicitly inferred target architecture)

Full log: 
[12.txt](https://github.com/dotnet/runtime/files/4406390/12.txt)

Seems like it gets LLVM NuGet for wrong platform:
```
2020-03-31T01:16:16.6445927Z   make[2]: Entering directory '/home/ubuntu/agent/_work/1/s/work/runtime/artifacts/obj/mono/Linux.arm64.Release/llvm'
2020-03-31T01:16:16.6450102Z   /home/ubuntu/agent/_work/1/s/work/runtime/src/mono/llvm/build_llvm_config.sh "/home/ubuntu/agent/_work/1/s/work/runtime/artifacts/obj/mono/Linux.arm64.Release/llvm/bin/llvm-config" "aarch64codegen" "-lz" | tee /home/ubuntu/agent/_work/1/s/work/runtime/artifacts/obj/mono/Linux.arm64.Release/llvm/llvm_config.mk
2020-03-31T01:16:16.6539102Z   /home/ubuntu/agent/_work/1/s/work/runtime/src/mono/llvm/build_llvm_config.sh: line 47: /home/ubuntu/agent/_work/1/s/work/runtime/artifacts/obj/mono/Linux.arm64.Release/llvm/bin/llvm-config: cannot execute binary file: Exec format error
2020-03-31T01:16:16.6552234Z   /home/ubuntu/agent/_work/1/s/work/runtime/src/mono/llvm/build_llvm_config.sh: line 47: 0: command not found
2020-03-31T01:16:16.6585163Z   /home/ubuntu/agent/_work/1/s/work/runtime/src/mono/llvm/build_llvm_config.sh: line 48: /home/ubuntu/agent/_work/1/s/work/runtime/artifacts/obj/mono/Linux.arm64.Release/llvm/bin/llvm-config: cannot execute binary file: Exec format error
2020-03-31T01:16:16.6620320Z   /home/ubuntu/agent/_work/1/s/work/runtime/src/mono/llvm/build_llvm_config.sh: line 49: /home/ubuntu/agent/_work/1/s/work/runtime/artifacts/obj/mono/Linux.arm64.Release/llvm/bin/llvm-config: cannot execute binary file: Exec format error
2020-03-31T01:16:16.6659986Z   /home/ubuntu/agent/_work/1/s/work/runtime/src/mono/llvm/build_llvm_config.sh: line 50: /home/ubuntu/agent/_work/1/s/work/runtime/artifacts/obj/mono/Linux.arm64.Release/llvm/bin/llvm-config: cannot execute binary file: Exec format error
2020-03-31T01:16:16.6702449Z   /home/ubuntu/agent/_work/1/s/work/runtime/src/mono/llvm/build_llvm_config.sh: line 51: /home/ubuntu/agent/_work/1/s/work/runtime/artifacts/obj/mono/Linux.arm64.Release/llvm/bin/llvm-config: cannot execute binary file: Exec format error
2020-03-31T01:16:16.6798347Z   /home/ubuntu/agent/_work/1/s/work/runtime/src/mono/llvm/build_llvm_config.sh: line 60: /home/ubuntu/agent/_work/1/s/work/runtime/artifacts/obj/mono/Linux.arm64.Release/llvm/bin/llvm-config: cannot execute binary file: Exec format error
```

Almost there... :D I assume I don't build with the exact same compiler version:

```
2020-03-31T14:40:05.2830774Z   /usr/bin/../lib/gcc/aarch64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/bits/basic_string.h:4323: undefined reference to `MonoEHFrameSymbol'
2020-03-31T14:40:05.2834120Z   /usr/bin/../lib/gcc/aarch64-linux-gnu/7.5.0/../../../../include/c++/7.5.0/bits/basic_string.h:4323: undefined reference to `MonoEHFrameSymbol'
2020-03-31T14:40:05.2912169Z   ./.libs/libmini.a(llvm-jit.o): In function `llvm::RTDyldMemoryManager::getSymbolAddress(std::string const&)':
2020-03-31T14:40:05.2916631Z   /home/ubuntu/agent/_work/1/s/work/runtime/artifacts/obj/mono/Linux.arm64.Release/llvm/include/llvm/ExecutionEngine/RTDyldMemoryManager.h:84: undefined reference to `llvm::RTDyldMemoryManager::getSymbolAddressInProcess(std::string const&)'
2020-03-31T14:40:05.2920858Z   ./.libs/libmini.a(llvm-jit.o):(.data.rel.ro+0x90): undefined reference to `llvm::RTDyldMemoryManager::getPointerToNamedFunction(std::string const&, bool)'
2020-03-31T14:40:05.9000698Z   clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Nevertheless I can confirm that the fix in #34322 works and downloads the correct NuGet package now.

--- a/docs/workflow/building/mono/README.md
+++ b/docs/workflow/building/mono/README.md
@@ -30,6 +30,11 @@ For both `build.sh` and `mono.sh`
 
 `/p:MonoEnableLlvm=true` - Builds mono w/ LLVM
 
+`/p:MonoEnableLlvm=true /p:MonoLLVMDir=path/to/llvm` - Builds mono w/ LLVM from a custom path
+
+`/p:MonoEnableLlvm=true /p:MonoLLVMDir=path/to/llvm /p:MonoLLVMUseCxx11Abi=true` - Builds mono w/ LLVM 
+from a custom path (and that LLVM was built with C++11 ABI)
+
 For `build.sh`
 
 `/p:DisableCrossgen=true` - Skips building the installer if you don't need it (builds faster)

--- a/src/mono/mono.proj
+++ b/src/mono/mono.proj
@@ -61,7 +61,7 @@
       <_MonoCXXFLAGS Include="-O0" />
       <_MonoCXXFLAGS Include="-ggdb3" />
       <_MonoCXXFLAGS Include="-fno-omit-frame-pointer" />
-      <_MonoCXXFLAGS Condition="$(HostArch.Contains('arm')) and '$(MonoEnableLLVM)' == 'true'" Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
+      <_MonoCXXFLAGS Condition="!$(HostArch.Contains('arm')) or '$(MonoEnableLLVM)' != 'true'" Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
     </ItemGroup>
 
     <!-- Release specific options -->
@@ -71,7 +71,7 @@
 
       <_MonoCXXFLAGS Include="-O2" />
       <_MonoCXXFLAGS Include="-g" />
-      <_MonoCXXFLAGS Condition="$(HostArch.Contains('arm')) and '$(MonoEnableLLVM)' == 'true'" Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
+      <_MonoCXXFLAGS Condition="!$(HostArch.Contains('arm')) or '$(MonoEnableLLVM)' != 'true'" Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
     </ItemGroup>
 
     <!-- iOS device specific options -->

--- a/src/mono/mono.proj
+++ b/src/mono/mono.proj
@@ -71,7 +71,7 @@
 
       <_MonoCXXFLAGS Include="-O2" />
       <_MonoCXXFLAGS Include="-g" />
-      <_MonoCXXFLAGS Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
+      <_MonoCXXFLAGS Condition="$(HostArch.Contains('arm')) and '$(MonoEnableLLVM)' == 'true'" Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
     </ItemGroup>
 
     <!-- iOS device specific options -->

--- a/src/mono/mono.proj
+++ b/src/mono/mono.proj
@@ -61,7 +61,6 @@
       <_MonoCXXFLAGS Include="-O0" />
       <_MonoCXXFLAGS Include="-ggdb3" />
       <_MonoCXXFLAGS Include="-fno-omit-frame-pointer" />
-      <_MonoCXXFLAGS Condition="'$(HostArch)' == 'x64' and '$(MonoEnableLLVM)' == 'true'" Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
     </ItemGroup>
 
     <!-- Release specific options -->
@@ -71,12 +70,11 @@
 
       <_MonoCXXFLAGS Include="-O2" />
       <_MonoCXXFLAGS Include="-g" />
-      <_MonoCXXFLAGS Condition="!$(HostArch.Contains('arm')) or '$(MonoEnableLLVM)' != 'true'" Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
     </ItemGroup>
 
-    <ItemGroup>
-      <!-- We build LLVM bits for x64 without C++11 ABI -->
-      <_MonoCXXFLAGS Condition="'$(HostArch)' == 'x64' and '$(MonoEnableLLVM)' == 'true' and '$(MonoLLVMUseCxx11Abi)' != 'true'" Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
+    <!-- We build LLVM bits for x64 without C++11 ABI -->
+    <ItemGroup Condition="'$(HostArch)' == 'x64' and '$(MonoEnableLLVM)' == 'true' and '$(MonoLLVMUseCxx11Abi)' != 'true'">
+      <_MonoCXXFLAGS Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
     </ItemGroup>
 
     <!-- iOS device specific options -->

--- a/src/mono/mono.proj
+++ b/src/mono/mono.proj
@@ -61,7 +61,7 @@
       <_MonoCXXFLAGS Include="-O0" />
       <_MonoCXXFLAGS Include="-ggdb3" />
       <_MonoCXXFLAGS Include="-fno-omit-frame-pointer" />
-      <_MonoCXXFLAGS Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
+      <_MonoCXXFLAGS Condition="$(HostArch.Contains('arm')) and '$(MonoEnableLLVM)' == 'true'" Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
     </ItemGroup>
 
     <!-- Release specific options -->

--- a/src/mono/mono.proj
+++ b/src/mono/mono.proj
@@ -61,7 +61,7 @@
       <_MonoCXXFLAGS Include="-O0" />
       <_MonoCXXFLAGS Include="-ggdb3" />
       <_MonoCXXFLAGS Include="-fno-omit-frame-pointer" />
-      <_MonoCXXFLAGS Condition="!$(HostArch.Contains('arm')) or '$(MonoEnableLLVM)' != 'true'" Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
+      <_MonoCXXFLAGS Condition="'$(HostArch)' == 'x64' and '$(MonoEnableLLVM)' == 'true'" Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
     </ItemGroup>
 
     <!-- Release specific options -->
@@ -72,6 +72,11 @@
       <_MonoCXXFLAGS Include="-O2" />
       <_MonoCXXFLAGS Include="-g" />
       <_MonoCXXFLAGS Condition="!$(HostArch.Contains('arm')) or '$(MonoEnableLLVM)' != 'true'" Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <!-- We build LLVM bits for x64 without C++11 ABI -->
+      <_MonoCXXFLAGS Condition="'$(HostArch)' == 'x64' and '$(MonoEnableLLVM)' == 'true' and '$(MonoLLVMUseCxx11Abi)' != 'true'" Include="-D_GLIBCXX_USE_CXX11_ABI=0" />
     </ItemGroup>
 
     <!-- iOS device specific options -->

